workers/datasets_based/src/datasets_based/workers/split_names.py

```diff
@@ -0,0 +1,132 @@
+    """Raised when the dataset has no data."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "EmptyDatasetError", cause, True)
+
+
+class SplitNameItem(TypedDict):
+    dataset: str
+    config: str
+    split: str
+
+
+class SplitNamesResponseContent(TypedDict):
+    split_names: List[SplitNameItem]
+
+
+def compute_split_names_response(
+    dataset: str,
+    config: str,
+    hf_token: Optional[str] = None,
+) -> SplitNamesResponseContent:
+    """
+    Get the response of /split-names for one specific dataset and config on huggingface.co.
+    Dataset can be private or gated if you pass an acceptable token.
+
+    It is assumed that the dataset exists and can be accessed using the token, and that the config exists in
+    the dataset.
+
+    This function relies on the streaming mode if the splits are not directly defined in the dataset config. See
+    https://github.dev/huggingface/datasets/blob/e183a269067575db8765ee979bd8523d14a1adae/src/datasets/inspect.py#L389-L390
+
+    The /split-names response generated by this function does not include stats about the split, like the size or
+    number of samples. See /dataset-info or /sizes for that.
+
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        config (`str`):
+            A configuration name.
+        hf_token (`str`, *optional*):
+            An authentication token (See https://huggingface.co/settings/token)
+    Returns:
+        `SplitNamesResponseContent`: An object with the list of split names for the dataset and config.
+    <Tip>
+    Raises the following errors:
+        - [`~workers.splits.EmptyDatasetError`]
+          The dataset is empty.
+        - [`~workers.splits.SplitsNamesError`]
+          If the list of splits could not be obtained using the datasets library.
+    </Tip>
+    """
+    logging.info(f"get splits for dataset={dataset}, config={config}")
+    use_auth_token: Union[bool, str, None] = hf_token if hf_token is not None else False
+    # get the list of splits in streaming mode
+    try:
+        split_name_items: List[SplitNameItem] = [
+            {"dataset": dataset, "config": config, "split": str(split)}
+            for split in get_dataset_split_names(path=dataset, config_name=config, use_auth_token=use_auth_token)
+        ]
+    except _EmptyDatasetError as err:
+        raise EmptyDatasetError("The dataset is empty.", cause=err) from err
+    except Exception as err:
+        raise SplitNamesError("Cannot get the split names for the dataset and config.", cause=err) from err
+    return {"split_names": split_name_items}
+
+
+class SplitNamesWorker(DatasetsBasedWorker):
+    @staticmethod
+    def get_job_type() -> str:
+        return "/split-names"
+
+    @staticmethod
+    def get_version() -> str:
+        return "2.0.0"
```

Comment:
I am not sure but since this is a new route, why it is not 1.0.0?

Reply:
copy/paste from split.py, and I forgot to change. Thanks